infra/lib/cleanup-users.ts

```diff
@@ -10,33 +10,35 @@ const { USER_POOL_ID } = process.env;
 export const handler: lambda.Handler = async (event) => {
     console.log(`Got event: ${JSON.stringify(event, null, 2)}`);
 
-    const users: UserType[] = [];
     try {
         console.log('Listing users...');
 
-        let nextToken: string | undefined;
-        do {
+        while (true) {
             const usersResp = await CLIENT.listUsers({
                 UserPoolId: USER_POOL_ID!,
-                PaginationToken: nextToken,
             });
-            for (const user of usersResp.Users ?? []) {
-                users.push(user);
+            const users: UserType[] = usersResp.Users ?? [];
+            if (users.length === 0) {
+                break;
             }
-            nextToken = usersResp.PaginationToken;
-        } while (nextToken);
-    } catch (e) {
-        console.error(`Error listing users: ${e}`);
-        return;
-    }
 
-    console.log(`Got users: ${JSON.stringify(users, null, 2)}`);
+            console.log(`Got users: ${JSON.stringify(users, null, 2)}`);
+            await Promise.all(
+                users.map(async (user) => {
```

Comment:
How many users would be listed in one batch, do we need to add throttling within each batch to avoid `TooManyRequestsException`?

Reply:
One batch is like 100 users I think. No need to throttle the batch.